src/modules/Profile/lib/Profile/Api/User.php

```diff
@@ -640,4 +640,143 @@ public function encodeurl($args)
         // construct the custom url part
         return $args['modname'] . '/' . $args['func'] . '/' . $vars;
     }
+
+    /**
+     * Get module links
+     *
+     * @return array array of links
+     */
+    public function getLinks()
+    {
+        if (!SecurityUtil::checkPermission('Profile::', '::', ACCESS_OVERVIEW)) {
```

Comment:
@nmpetkov This permission rule wouldn't be valid for all of the links that this function provides, since there are links to other modules such as ContactList and Avatar. I think permissions should be checked per link, based on the relevant condition. Not only checking to see if those other modules are available, but also checking if the user has access to those particular modules too.